bundles/org.openhab.binding.anthem/src/main/java/org/openhab/binding/anthem/internal/handler/AnthemHandler.java

```diff
@@ -0,0 +1,440 @@
+        localSenderThread.setDaemon(true);
+        localSenderThread.start();
+        this.senderThread = localSenderThread;
+
+        updateStatus(ThingStatus.ONLINE);
+
+        ScheduledFuture<?> localPollingJob = this.pollingJob;
+        if (localPollingJob == null) {
+            this.pollingJob = scheduler.scheduleWithFixedDelay(this::poll, POLLING_DELAY_SECONDS,
+                    POLLING_INTERVAL_SECONDS, TimeUnit.SECONDS);
+        }
+    }
+
+    private void poll() {
+        logger.debug("Polling...");
+        sendCommand(AnthemCommand.queryPower(Zone.MAIN));
+        sendCommand(AnthemCommand.queryPower(Zone.ZONE2));
+    }
+
+    private void scheduleConnectRetry(long waitMinutes) {
+        logger.debug("Scheduling connection retry in {} minutes", waitMinutes);
+        connectRetryJob = scheduler.schedule(this::connect, waitMinutes, TimeUnit.MINUTES);
+    }
+
+    private synchronized void disconnect() {
+        logger.debug("Disconnecting from Anthem");
+
+        ScheduledFuture<?> localPollingJob = this.pollingJob;
+        if (localPollingJob != null && !localPollingJob.isDone()) {
+            localPollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+
+        ScheduledFuture<?> localConnectRetryJob = this.connectRetryJob;
+        if (localConnectRetryJob != null) {
+            localConnectRetryJob.cancel(true);
+            this.connectRetryJob = null;
+        }
+
+        Thread localSenderThread = this.senderThread;
+        if (localSenderThread != null && localSenderThread.isAlive()) {
+            localSenderThread.interrupt();
+        }
+
+        Thread localReaderThread = this.readerThread;
+        if (localReaderThread != null && localReaderThread.isAlive()) {
+            localReaderThread.interrupt();
+        }
+        Socket localSocket = this.socket;
+        if (localSocket != null) {
+            try {
+                localSocket.close();
+            } catch (IOException e) {
+                logger.debug("Error closing socket: {}", e.getMessage());
+            }
+            this.socket = null;
+        }
+        BufferedReader localReader = this.reader;
+        if (localReader != null) {
+            try {
+                localReader.close();
+            } catch (IOException e) {
+                logger.debug("Error closing reader: {}", e.getMessage());
+            }
+            this.reader = null;
+        }
+        BufferedWriter localWriter = this.writer;
+        if (localWriter != null) {
+            try {
+                localWriter.close();
+            } catch (IOException e) {
+                logger.debug("Error closing writer: {}", e.getMessage());
+            }
+            this.writer = null;
+        }
+    }
+
+    private synchronized void reconnect() {
+        logger.debug("Attempting to reconnect to the Anthem");
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
```

Comment:
It seems strange to set "communication error" before attempting a reconnect. If we are not already connected, I'd expect status **offline** already, so no need to set status.

Reply:
Ok